rest/route/host_agent.go

```diff
@@ -483,7 +483,7 @@ func assignNextAvailableTask(ctx context.Context, env evergreen.Environment, tas
 
 		var queueItem *model.TaskQueueItem
 		switch d.DispatcherSettings.Version {
-		case evergreen.DispatcherVersionRevised, evergreen.DispatcherVersionRevisedWithDependencies:
```

Comment:
GitHub doesn't let me comment on unchanged lines, but is the `default` case below supposed to error? It was only used for the legacy implementation.

Similarly, could we remove `(TaskQueueItemDispatcher).FindNextTask` since it seems like it's not used anywhere except in the default below?

Reply:
Ah sure I can update that